src/main/java/dev/warrant/WarrantClient.java

```diff
@@ -199,17 +194,42 @@ public String createSelfServiceSession(Session session) throws WarrantException
         }
     }
 
+    public WarrantCheckResult isAuthorized(WarrantCheck toCheck) throws WarrantException {
+        String op = toCheck.getOp();
+        if (op == null || op.isEmpty()) {
+            throw new WarrantException("Must include a valid op");
+        }
+        if (!"allOf".equals(op) && !"anyOf".equals(op)) {
+            throw new WarrantException("Invalid op " + op);
+        }
+        HttpResponse<String> resp = makePostRequest("/v2/authorize", toCheck);
+        if (resp.statusCode() != Response.Status.OK.getStatusCode()) {
+            throw new WarrantException("API error: " + resp.statusCode());
+        }
+        try {
+            WarrantCheckResult result = mapper.readValue(resp.body(), WarrantCheckResult.class);
+            return result;
+        } catch (IOException e) {
+            throw new WarrantException(e);
+        }
+    }
+
     public boolean hasPermission(String permissionId, String userId) throws WarrantException {
         Subject userSubject = new Subject("user", userId);
         Warrant permissionWarrant = new Warrant("permission", permissionId, "member", userSubject);
-        return isAuthorized(permissionWarrant);
+        WarrantCheck permissionCheck = new WarrantCheck(Arrays.asList(permissionWarrant), "allOf");
```

Comment:
Not sure if it was intentional but for single warrants we can omit the `op` param